src/contexts/wallet.tsx

```diff
@@ -194,25 +215,116 @@ export const WalletProvider = ({ children = null as any }) => {
+    setTxHash(signed_restore_tx.hash().toString('hex'));
+    let restore_tx_resp = await sendTransaction(signed_restore_tx, (result: string) => undefined);
+    if (restore_tx_resp) {
+      return restore_tx_resp;
+    } else {
+      throw new Error('Restore transaction failed');
+    }
+  }
+
+  async function sendTransaction<T>(
+    transaction: Transaction,
+    parser: (result: string) => T
+  ): Promise<ContractResponse<T> | undefined> {
+    let send_tx_response = await rpc.sendTransaction(transaction);
+    let curr_time = Date.now();
+    while (send_tx_response.status !== 'PENDING' && Date.now() - curr_time < 5000) {
+      await new Promise((resolve) => setTimeout(resolve, 1000));
+      send_tx_response = await rpc.sendTransaction(transaction);
+    }
+    if (send_tx_response.status !== 'PENDING') {
+      setFailureMessage(`Transaction failed to submit with, ${send_tx_response.status}`);
+      setTxStatus(TxStatus.FAIL);
+      return;
+    }
+
+    let get_tx_response = await rpc.getTransaction(send_tx_response.hash);
+    while (get_tx_response.status === 'NOT_FOUND') {
+      await new Promise((resolve) => setTimeout(resolve, 1000));
+      get_tx_response = await rpc.getTransaction(send_tx_response.hash);
+    }
+    let tx_resp = ContractResponse.fromTransactionResponse(
+      get_tx_response,
+      transaction,
+      network.passphrase,
+      parser
+    );
+    setTxHash(tx_resp.hash);
+    if (tx_resp.result.isOk()) {
+      console.log('Successfully submitted transaction: ', tx_resp.hash);
+      setTxStatus(TxStatus.SUCCESS);
+    } else {
+      console.log('Failed submitted transaction: ', tx_resp.hash);
+      setFailureMessage(tx_resp.result.unwrapErr().message);
+      setTxStatus(TxStatus.FAIL);
+    }
+    return tx_resp;
+  }
+
   async function submitTransaction<T>(
-    submission: Promise<ContractResult<T>>,
+    operation: xdr.Operation,
+    parser: (result: string) => T,
+    sim: boolean,
     poolId?: string | undefined
-  ): Promise<T | undefined> {
+  ): Promise<ContractResponse<T> | undefined> {
     try {
       // submission calls `sign` internally which handles setting TxStatus
+      let account = await rpc.getAccount(walletAddress);
+      let tx_builder = new TransactionBuilder(account, {
+        networkPassphrase: network.passphrase,
+        fee: BASE_FEE,
+        timebounds: { minTime: 0, maxTime: Math.floor(Date.now() / 1000) + 5 * 60 * 1000 },
+      }).addOperation(operation);
+      let transaction = tx_builder.build();
+
+      let simulation = await rpc.simulateTransaction(transaction);
+      let sim_response = ContractResponse.fromSimulationResponse(
+        simulation,
+        transaction,
+        network.passphrase,
+        parser
+      );
+      if (sim) {
+        return sim_response;
+      }
+      if (sim_response.result.isErr()) {
+        if (SorobanRpc.Api.isSimulationRestore(simulation)) {
+          let response = await restore(simulation, transaction);
```

Comment:
we probably don't want to automatically restore. The UX on the user submission side will be weird. There will basically be no UI interaction and a vastly different XDR.

I think we need to rethink the UX for this, and likely requires a followup PR. For now, could we at least remove this restore call and set some failure message here?

Reply:
This also likely requires some changes made in the followup PR to do correctly

Reply:
Agreed that this will require some UX work. When restoring is needed we could have a screen popup displaying that an entry has expired and then have a bottom confirming that the user wants to restore. After restoring the normal transaction appears.

Reply:
Going to remove this for the time being